en/middleware/mavlink.md

```diff
@@ -519,3 +525,55 @@ However if you use your own dialect then it should include ArduPilotMega.xml (or
 ### Updating MAVSDK
 
 See the MAVSDK docs for information about how to work with [MAVLink headers and dialects](https://mavsdk.mavlink.io/main/en/cpp/guide/build.html).
+
+### Updating MAVProxy
+
+[MAVProxy](https://ardupilot.org/mavproxy/) is UAV ground station software package for MAVLink based systems written in Python.
+While not optimized for PX4 or widely promoted in the PX4 community, we mention it here because it is one of the easiest ground stations to update with custom messages, and is therefore one of the easiest ways to test that your custom messages are being sent.
+
+[MAVProxy](https://ardupilot.org/mavproxy/) depends on [Pymavlink](https://mavlink.io/en/mavgen_python/).
+The approach described below is to install MAVProxy, remove the default-installed Pymavlink, and then rebuild Pymavlink with the new messages.
+
+The steps are:
+
+1. Install MAVproxy using the command:
+
+   ```sh
+   python -m pip install mavproxy
+   ```
+
+1. Remove the Pymavlink package installed with MAVProxy:
+
+   ```sh
+   python -m pip uninstall pymavlink
+   ```
+
+1. Clone the MAVLink repository (which also contains Pymavlink)
+
+   ```sh
+   git clone https://github.com/mavlink/mavlink.git --recursive
+   cd mavlink
+   ```
+
+1. Copy your custom message XML into the [message_definitions/v1.0](https://github.com/mavlink/mavlink/tree/master/message_definitions/v1.0) folder
+1. Generate the headers:
+
+   ```sh
+   python3 -m pymavlink.tools.mavgen --lang=C --wire-protocol=2.0 --output=generated message_definitions/v1.0/your_dialect.xml
+   ```
+
+1. Install Pymavlink using
+
+   ```sh
+   cd pymavlink
+   sudo python setup.py install
```

Comment:
I am not 100% certain this will work. Normally Pymavlink package is built with the ArduPilot fork of mavlink/mavlink headers.
What makes me think it probably does work is:

- The developer instructions https://github.com/ArduPilot/pymavlink?tab=readme-ov-file#for-developers  indicate that by default if you cloned MAVLink the location of the headers comes from the parent directoy.
- I assume you tested this :-)

Reply:
@hamishwillee yes I have tested it

Reply:
Running this with `sudo` is asking to mess up your Python dependencies. I don't think that's something we should have in the docs.

What are we trying to do here? And why can't we just use upstream pymavlink?

Reply:
Yes, this does indeed screw up your installation - as I have just verified. I can get this to "work" (somewhat) but I can't undo it. Will have to delete my installation and start again :-(

What we are trying to do, ultimately, is have an easy way to test a custom message, since building QGC with a custom dialect is a bit of a pain. 

The solution described here is to install mavproxy and then delete the pymavlink that it depends on. Then rebuild pymavlink from the mavlink repo clone with your new xml definitions added. This step is easy because you just copy the xml files in and run setup.py and it creates a new installation/.egg with the .py files for the definitions.
Mavproxy can then display the message if it is received.

1. We definitely don't need the step above building the files from the definitions. the setup.py does that for you.
2. This still doesn't work though due to the error MAV_TYPE_VTOL_DUOROTOR not defined. That is correct - it has been removed from Mavlink, but it seems that mavproxy expects it in https://github.com/ArduPilot/MAVProxy/blob/master/MAVProxy/modules/mavproxy_link.py#L744
   - @wasaybaig How did you work your way around that? It's a bug in mavproxy.
3. I presume we can completely avoid all use of sudo if we install mavproxy and run setup from inside a virtual env?

Reply:
FYI, let's see if we can get MAVProxy to update https://github.com/ArduPilot/MAVProxy/issues/1470